### DISCUSSION TITLE
Adding status code to the maps in json format

### DISCUSSION THREAD

**Discussion by vilben** [lycheeverse/lychee#1013] [Ideas] [answered]
Currently, when running lychee with the output format=json, a fail_map, or success_map includes objects with an url and a status as a human readable text.

command:
`cargo run -- www.example.com --format json -v `
output: 
```
{
  "detailed_stats": true,
  ...
  "success_map": {
    "http://www.example.com/": [
      {
        "url": "https://www.iana.org/domains/example",
        "status": "OK (200 OK)"
      }
    ]
  },
 ...
}
```
Instead of having the status as a readable string, it might be useful in some cases to have the actual error code.

For example:
```
{
  "detailed_stats": true,
  ...
  "success_map": {
      "http://www.example.com/": [
        {
          "url": "https://www.iana.org/domains/example",
          "status": {
            "text": "OK",
            "code": 200
          }
        }
      ]
    }
 ...
}
```

What do you think about that?

**Comment by mre** [comment, marked as answer]
Good idea. I'd accept a pull request for that.
